crates/zizmor/src/audit/use_trusted_publishing.rs

```diff
@@ -356,6 +427,21 @@ impl Audit for UseTrustedPublishing {
             });
 
             for subfeature in self.trusted_publishing_command_candidates(run, shell)? {
+                // If this looks like cargo publish and the job wires in a token
+                // from rust-lang/crates-io-auth-action, treat it as Trusted Publishing
+                // and don't emit a finding.
+                let skip_for_crates_io_tp = match &subfeature.fragment {
+                    subfeature::Fragment::Raw(cmd) => {
+                        Self::is_cargo_publish_command(cmd)
+                            && self.uses_crates_io_trusted_publishing(step, run)
+                    }
+                    _ => false,
+                };
+
+                if skip_for_crates_io_tp {
+                    continue;
+                }
```

Comment:
Hmm -- I think we'd ideally skip these lower-level checks entirely, i.e. we should probably remove `cargo publish` from `self.trusted_publishing_command_candidates` now that it's specialized like this.